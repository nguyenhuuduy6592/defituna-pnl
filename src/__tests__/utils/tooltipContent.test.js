import {
  getFeeAPRTooltip,
  getVolumeTVLTooltip,
  getVolatilityTooltip,
  getTVLTooltip,
  getVolumeTooltip,
  getYieldTooltip,
  getFeeRateTooltip,
  getImpermanentLossTooltip,
  getPriceImpactTooltip,
} from '@/utils/tooltipContent';

describe('Tooltip Content Utilities', () => {
  describe('getFeeAPRTooltip', () => {
    it('should return correct tooltip for high APR', () => {
      expect(getFeeAPRTooltip('55.5')).toMatchSnapshot();
    });

    it('should return correct tooltip for medium-high APR', () => {
      expect(getFeeAPRTooltip('25')).toMatchSnapshot();
    });

    it('should return correct tooltip for medium APR', () => {
      expect(getFeeAPRTooltip('15')).toMatchSnapshot();
    });

    it('should return correct tooltip for low APR', () => {
      expect(getFeeAPRTooltip('5')).toMatchSnapshot();
    });

    it('should handle zero APR', () => {
      expect(getFeeAPRTooltip('0')).toMatchSnapshot();
    });

    it('should handle non-numeric input gracefully (defaults to low)', () => {
      // parseFloat('abc') results in NaN, which fails all checks >= 10
      expect(getFeeAPRTooltip('abc')).toMatchSnapshot();
    });
  });

  describe('getVolumeTVLTooltip', () => {
    it('should return correct tooltip for excellent efficiency', () => {
      expect(getVolumeTVLTooltip('6')).toMatchSnapshot();
    });

    it('should return correct tooltip for good efficiency', () => {
      expect(getVolumeTVLTooltip('3')).toMatchSnapshot();
    });

    it('should return correct tooltip for average efficiency', () => {
      expect(getVolumeTVLTooltip('1.5')).toMatchSnapshot();
    });

    it('should return correct tooltip for below average efficiency', () => {
      expect(getVolumeTVLTooltip('0.7')).toMatchSnapshot();
    });

    it('should return correct tooltip for poor efficiency', () => {
      expect(getVolumeTVLTooltip('0.2')).toMatchSnapshot();
    });

     it('should handle zero ratio', () => {
      expect(getVolumeTVLTooltip('0')).toMatchSnapshot();
    });

    it('should handle non-numeric input gracefully (defaults to poor)', () => {
       // parseFloat('abc') results in NaN, which fails all checks >= 0.5
      expect(getVolumeTVLTooltip('abc')).toMatchSnapshot();
    });
  });

  describe('getVolatilityTooltip', () => {
    it('should return correct tooltip for high volatility', () => {
      expect(getVolatilityTooltip('High')).toMatchSnapshot();
    });

     it('should return correct tooltip for medium volatility (case insensitive)', () => {
      expect(getVolatilityTooltip('medium')).toMatchSnapshot();
    });

     it('should return correct tooltip for low volatility', () => {
      expect(getVolatilityTooltip('Low')).toMatchSnapshot();
    });

     it('should return default tooltip for unknown volatility', () => {
      expect(getVolatilityTooltip('unknown')).toMatchSnapshot();
    });

     it('should return default tooltip for empty input', () => {
      expect(getVolatilityTooltip('')).toMatchSnapshot();
    });
  });

  describe('getTVLTooltip', () => {
    it('should return the correct static tooltip content', () => {
      expect(getTVLTooltip()).toMatchSnapshot(); // Note: parameter 'value' is unused in implementation
    });
  });

  describe('getVolumeTooltip', () => {
    it('should return the correct tooltip content with timeframe', () => {
      expect(getVolumeTooltip('24h')).toMatchSnapshot();
    });
  });

  describe('getYieldTooltip', () => {
     it('should return the correct tooltip content with timeframe', () => {
      expect(getYieldTooltip('7d')).toMatchSnapshot();
    });
  });

  describe('getFeeRateTooltip', () => {
    it('should return the correct static tooltip content', () => {
      expect(getFeeRateTooltip()).toMatchSnapshot();
    });
  });

  describe('getImpermanentLossTooltip', () => {
    it('should return the correct static tooltip content', () => {
      expect(getImpermanentLossTooltip()).toMatchSnapshot();
    });
  });

  describe('getPriceImpactTooltip', () => {
    it('should return the correct static tooltip content', () => {
      expect(getPriceImpactTooltip()).toMatchSnapshot();
    });
  });
}); 