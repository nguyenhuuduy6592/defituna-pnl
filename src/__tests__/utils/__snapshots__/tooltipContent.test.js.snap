// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tooltip Content Utilities getFeeAPRTooltip should handle non-numeric input gracefully (defaults to low) 1`] = `
"Annual rate of return from trading fees if current activity continues.

Risk level: Low
More stable but lower returns

• Consider other pools unless seeking stability
• Compare against other investments
• Separate from impermanent loss"
`;

exports[`Tooltip Content Utilities getFeeAPRTooltip should handle zero APR 1`] = `
"Annual rate of return from trading fees if current activity continues.

Risk level: Low
More stable but lower returns

• Consider other pools unless seeking stability
• Compare against other investments
• Separate from impermanent loss"
`;

exports[`Tooltip Content Utilities getFeeAPRTooltip should return correct tooltip for high APR 1`] = `
"Annual rate of return from trading fees if current activity continues.

Risk level: High
May not be sustainable long-term

• Check volatility before providing liquidity
• Compare against other investments
• Separate from impermanent loss"
`;

exports[`Tooltip Content Utilities getFeeAPRTooltip should return correct tooltip for low APR 1`] = `
"Annual rate of return from trading fees if current activity continues.

Risk level: Low
More stable but lower returns

• Consider other pools unless seeking stability
• Compare against other investments
• Separate from impermanent loss"
`;

exports[`Tooltip Content Utilities getFeeAPRTooltip should return correct tooltip for medium APR 1`] = `
"Annual rate of return from trading fees if current activity continues.

Risk level: Medium
Balanced fee generation

• Reasonable for the risk
• Compare against other investments
• Separate from impermanent loss"
`;

exports[`Tooltip Content Utilities getFeeAPRTooltip should return correct tooltip for medium-high APR 1`] = `
"Annual rate of return from trading fees if current activity continues.

Risk level: Medium-High
Strong fee generation

• Monitor volatility
• Compare against other investments
• Separate from impermanent loss"
`;

exports[`Tooltip Content Utilities getFeeRateTooltip should return the correct static tooltip content 1`] = `
"Percentage fee charged on every trade.

• Higher fees = Better for LPs (more per trade)
• Lower fees = Better for traders (lower costs)
• Ideal fee balances income vs. volume"
`;

exports[`Tooltip Content Utilities getImpermanentLossTooltip should return the correct static tooltip content 1`] = `
"Loss LPs face when token prices change vs. holding.

• Occurs when prices move from entry ratio
• "Impermanent" if prices return to original ratio
• More severe in volatile pairs

Risk Mitigation:
• Choose less volatile tokens
• Consider stablecoin pairs
• Balance fee income vs. potential IL"
`;

exports[`Tooltip Content Utilities getPriceImpactTooltip should return the correct static tooltip content 1`] = `
"Effect your trade has on market price.

• Larger trades = bigger impact
• Higher impact = worse execution price
• Related to pool depth and liquidity

Tips:
• Watch impact warnings
• Break large trades into smaller ones
• Higher TVL pools offer lower impact"
`;

exports[`Tooltip Content Utilities getTVLTooltip should return the correct static tooltip content 1`] = `
"Total value of assets deposited in this pool.

For traders:
• Higher TVL = better price execution
• Lower slippage for trades

For LPs:
• Higher TVL = more competition for fees
• Lower TVL = higher fee share but higher risk"
`;

exports[`Tooltip Content Utilities getVolatilityTooltip should return correct tooltip for high volatility 1`] = `
"How much token prices fluctuate relative to each other.

• High price fluctuation
• Significant IL risk
• Better for short-term positions

Lower volatility = more stable returns
Higher volatility = potential for higher fees but greater IL"
`;

exports[`Tooltip Content Utilities getVolatilityTooltip should return correct tooltip for low volatility 1`] = `
"How much token prices fluctuate relative to each other.

• Low price fluctuation
• Minimal IL risk
• Good for passive liquidity

Lower volatility = more stable returns
Higher volatility = potential for higher fees but greater IL"
`;

exports[`Tooltip Content Utilities getVolatilityTooltip should return correct tooltip for medium volatility (case insensitive) 1`] = `
"How much token prices fluctuate relative to each other.

• Moderate price fluctuation
• Moderate IL risk
• Balanced risk/reward

Lower volatility = more stable returns
Higher volatility = potential for higher fees but greater IL"
`;

exports[`Tooltip Content Utilities getVolatilityTooltip should return default tooltip for empty input 1`] = `
"How much token prices fluctuate relative to each other.

• Unknown volatility
• Unknown IL risk
• Exercise caution

Lower volatility = more stable returns
Higher volatility = potential for higher fees but greater IL"
`;

exports[`Tooltip Content Utilities getVolatilityTooltip should return default tooltip for unknown volatility 1`] = `
"How much token prices fluctuate relative to each other.

• Unknown volatility
• Unknown IL risk
• Exercise caution

Lower volatility = more stable returns
Higher volatility = potential for higher fees but greater IL"
`;

exports[`Tooltip Content Utilities getVolumeTVLTooltip should handle non-numeric input gracefully (defaults to poor) 1`] = `
"How efficiently the pool uses liquidity to generate volume.

Capital Efficiency: Poor
Significantly over-supplied

• Higher ratios = more trading per $ deposited
• Low ratios = reduced fee opportunities
• Helps identify underserved trading pairs"
`;

exports[`Tooltip Content Utilities getVolumeTVLTooltip should handle zero ratio 1`] = `
"How efficiently the pool uses liquidity to generate volume.

Capital Efficiency: Poor
Significantly over-supplied

• Higher ratios = more trading per $ deposited
• Low ratios = reduced fee opportunities
• Helps identify underserved trading pairs"
`;

exports[`Tooltip Content Utilities getVolumeTVLTooltip should return correct tooltip for average efficiency 1`] = `
"How efficiently the pool uses liquidity to generate volume.

Capital Efficiency: Average
Balanced

• Higher ratios = more trading per $ deposited
• Low ratios = reduced fee opportunities
• Helps identify underserved trading pairs"
`;

exports[`Tooltip Content Utilities getVolumeTVLTooltip should return correct tooltip for below average efficiency 1`] = `
"How efficiently the pool uses liquidity to generate volume.

Capital Efficiency: Below average
Possibly over-supplied

• Higher ratios = more trading per $ deposited
• Low ratios = reduced fee opportunities
• Helps identify underserved trading pairs"
`;

exports[`Tooltip Content Utilities getVolumeTVLTooltip should return correct tooltip for excellent efficiency 1`] = `
"How efficiently the pool uses liquidity to generate volume.

Capital Efficiency: Excellent
May need more liquidity

• Higher ratios = more trading per $ deposited
• Low ratios = reduced fee opportunities
• Helps identify underserved trading pairs"
`;

exports[`Tooltip Content Utilities getVolumeTVLTooltip should return correct tooltip for good efficiency 1`] = `
"How efficiently the pool uses liquidity to generate volume.

Capital Efficiency: Good
Well-utilized

• Higher ratios = more trading per $ deposited
• Low ratios = reduced fee opportunities
• Helps identify underserved trading pairs"
`;

exports[`Tooltip Content Utilities getVolumeTVLTooltip should return correct tooltip for poor efficiency 1`] = `
"How efficiently the pool uses liquidity to generate volume.

Capital Efficiency: Poor
Significantly over-supplied

• Higher ratios = more trading per $ deposited
• Low ratios = reduced fee opportunities
• Helps identify underserved trading pairs"
`;

exports[`Tooltip Content Utilities getVolumeTooltip should return the correct tooltip content with timeframe 1`] = `
"Total trading activity in this pool over 24h.

For traders:
• Higher volume = better liquidity & spreads

For LPs:
• Higher volume = more fees earned
• Volume trends help predict future fees"
`;

exports[`Tooltip Content Utilities getYieldTooltip should return the correct tooltip content with timeframe 1`] = `
"Percentage return from fees over 7d.

• Direct indicator of recent returns
• Does NOT account for impermanent loss
• Compare to volatility for risk-adjusted assessment"
`;
